src/app/features/courses/courses.component.html

```diff
@@ -3,29 +3,31 @@
 </ng-container>
 
 <ng-template #coursesTemplate>
-  <ng-container *ngIf="courses.length>0; else emptyList" >
+  <ng-container *ngIf="courses.length>0 || searching; else emptyList" >
     <div class="actionBar">
-      <app-search (search)="handleSearch($event)" [placeholder]="'Course...'"></app-search>
+      <app-search [placeholder]="'Course...'"></app-search>
       <ng-container *ngIf="isAdmin$ | async">
         <app-button (click)="onAddButtonClick()" [type]="ButtonTypes.Button" [buttonText]="'ADD NEW COURSE'"></app-button>
       </ng-container>
     </div>
 
-    <ng-container *ngIf="notFound">
-      <span [ngClass]="'noCourse'">No course found for {{lastSearchedText}} - All courses are shown</span>
-    </ng-container>
+    <div  [ngClass]="'noCourseContainer'">
```

Comment:
Why use `[]` syntax when there is no condition applied to the div element class list?
I think better add the `*ngIf` conditional here.